pkg/bff/auth/auth.go

```diff
@@ -202,6 +241,16 @@ func GetRegisteredClaims(c *gin.Context) (*validator.RegisteredClaims, error) {
 	if !exists {
 		return nil, ErrNoClaims
 	}
-	rclaims := claims.(validator.RegisteredClaims)
-	return &rclaims, nil
+	rclaims := claims.(*validator.RegisteredClaims)
+	return rclaims, nil
+}
+
+// GetUserInfo fetches the user info from the gin context. Returns an error if no user
+// exists on the context or if the user value is nil. Panics if user is incorrect type.
+func GetUserInfo(c *gin.Context) (*management.User, error) {
+	user, exists := c.Get(ContextUserInfo)
+	if !exists || user == nil {
+		return nil, ErrNoUserInfo
+	}
+	return user.(*management.User), nil
```

Comment:
That's a good question - in the docstring I mention that this function will cause a panic if the user is an incorrect type, and I think that's important here. I tend to differentiate between user errors (e.g. if the user can change the request and have it succeed) vs system or developer errors (that have to be changed in code). If this is the wrong type, it's the latter kind of error and we need to be alerted to it. 

The GDS BFF code has two things that deal with the panic:

1. panic recovery middleware that will return a 500 to the user 
2. sentry logging that will alert us to the panic and create a shortcut story for us 

So in this case, I'm very comfortable with allowing the panic. In the past, if we wanted to prevent the panic while still being able to track problems in production we would do something like this:

```
log.WithLevel(zerolog.Panic).Err(err).Str("type", fmt.Sprintf("%T", user).Msg("unexpected user type")
```

This would log the problem and send a slack message to our #gds-critical channel; but now that we have Sentry, just panicking is a bit easier.